main_js.go

```diff
@@ -728,13 +742,13 @@ func (pe *pcdeditor) runImpl(ctx context.Context) error {
 				gl.Uniform1ui(uMaxLabel, renderLabelMax)
 
 				gl.BindBuffer(gl.ARRAY_BUFFER, posBuf)
-				gl.VertexAttribPointer(aVertexPosition, 3, gl.FLOAT, false, pp.Stride(), 0)
-				gl.VertexAttribIPointer(aVertexLabel, 1, gl.UNSIGNED_INT, pp.Stride(), 3*4)
+				gl.VertexAttribPointer(aVertexPosition, 3, gl.FLOAT, false, pp.Stride()*samplingRatio, 0)
+				gl.VertexAttribIPointer(aVertexLabel, 1, gl.UNSIGNED_INT, pp.Stride()*samplingRatio, 3*4)
 				gl.UniformMatrix4fv(uModelViewMatrixLocation, false, modelViewMatrix)
 				gl.UniformMatrix4fv(uCropMatrixLocation, false, pe.cmd.CropMatrix())
 
 				gl.BindBuffer(gl.ARRAY_BUFFER, selectMaskBuf)
-				gl.VertexAttribIPointer(aSelectMask, 1, gl.UNSIGNED_INT, 4, 0)
+				gl.VertexAttribIPointer(aSelectMask, 1, gl.UNSIGNED_INT, 4*samplingRatio, 0)
```

Comment:
Is this correct here to multiply by `samplingRatio`?

Reply:
SelectMask is an array of the flags to keep that the each point is currently selected or not. It must be sampled with the same interval with the point position array to correctly color the selected points